### ISSUE TITLE
Not all folders are updated

### ISSUE THREAD

**Issue by Jolly-Pirate** [MestreLion/git-tools#47]
Usually `git restore-mtime` works on folder too, but in some cases it doesn't work on some folders in some repos. For example, here https://github.com/CodeBull/HiveAccountRecovery, the `css` and `js` folders.

**Comment by MestreLion** [comment]
Yes, the algorithm for updating directories has issues... I've listed a few thoughts about it at https://github.com/MestreLion/git-tools/blob/b3041998ceb4f92f144adc2226393ecc36e19f96/git-restore-mtime#L47 

The fix is to really rethink and rework directory updates, which is already planned but I don't have the time for it right now.

**Comment by MestreLion** [comment]
Fixing this would also properly fix #7

**Comment by mvdan** [comment]
Ran into this issue just now. Here's a workaround that's enough for my use case:

```
find . -not -path '*/.*' -type d -exec touch -t 202211302355 {} \; && git-restore-mtime
```

Essentially, I reset all directory modified times to a dummy timestamp in late 2022, and then run git-restore-mtime. The directories that git-restore-mtime does not update keep the dummy timestamp, but at least that's fixed and doesn't change from one `git clone` to another. That hack might be enough for others as well.
